libstuff/ResourceMonitorThread.h

```diff
@@ -0,0 +1,31 @@
+#include "libstuff/libstuff.h"
+#include <thread>
+
+using namespace std;
+
+// This class is a wrapper around the default thread. We use it to collect the thread CPU usage. That allows us
+// to investigate if we have any threads using more resources than it should, which can cause CPU usage peaks in
+// the cluster.
+class ResourceMonitorThread : public thread
+{
+public:
+    // When calling this constructor, if you're passing a class member function as the `f` parameter and that
+    // function receives parameters, you will need to wrap your function call in a lambda, doing something like:
+    // ResourceMonitorThread([=, this]{ this->memberFunction(param1, param2);});
+    template<typename F, typename... Args>
+    ResourceMonitorThread(F&& f, Args&&... args):
+      thread(ResourceMonitorThread::wrapper<F&&, Args&&...>, forward<F&&>(f), forward<Args&&>(args)...){};
+private:
+    static uint64_t threadStartTime;
+    static double cpuStartTime;
```

Comment:
These are broken now, they should not be static, the values are shared between threads that way. These should just be member variables.